SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,430 @@
+	FVector, Location);
+bool FSpawnLayer1PawnAtLocation::Update()
+{
+	FActorSpawnParameters SpawnParams;
+	SpawnParams.bNoFail = true;
+	SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+	AActor* NewActor = TestData->TestWorld->SpawnActor<ALayer1Pawn>(Location, FRotator::ZeroRotator, SpawnParams);
+	TestData->TestActors.Add(Handle, NewActor);
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FSpawnLayer2PawnAtLocation, TSharedPtr<TestData>, TestData,
+	FName, Handle, FVector, Location);
+bool FSpawnLayer2PawnAtLocation::Update()
+{
+	FActorSpawnParameters SpawnParams;
+	SpawnParams.bNoFail = true;
+	SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+	AActor* NewActor = TestData->TestWorld->SpawnActor<ALayer2Pawn>(Location, FRotator::ZeroRotator, SpawnParams);
+	TestData->TestActors.Add(Handle, NewActor);
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_FIVE_PARAMETER(FCheckActorsAuth, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, FName, FirstActorName, FName, SecondActorName, bool, ExpectEqual);
+bool FCheckActorsAuth::Update()
+{
+	const auto& Strat = TestData->Strat;
+	const auto& TestActors = TestData->TestActors;
+
+	const VirtualWorkerId FirstActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[FirstActorName]);
+	const VirtualWorkerId SecondActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[SecondActorName]);
+
+	if (ExpectEqual)
+	{
+		Test->TestEqual(
+			FString::Printf(TEXT("Actors should have the same auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+	}
+	else {
+		Test->TestNotEqual(
+			FString::Printf(TEXT("Actors should have different auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+	}
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckRequiresHandover, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, bool, Expected);
+bool FCheckRequiresHandover::Update()
+{
+	const bool Actual = TestData->Strat->RequiresHandoverData();
+	Test->TestEqual(
+		FString::Printf(TEXT("Strategy requires handover data Expected: %c Actual: %c"), Expected, Actual),
+		Expected, Actual);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckShouldHaveAuthMatchesWhoShouldHaveAuth, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, FName, ActorName);
+bool FCheckShouldHaveAuthMatchesWhoShouldHaveAuth::Update()
+{
+	const auto Strat = TestData->Strat;
+	const auto& TestActors = TestData->TestActors;
+
+	const bool WeShouldHaveAuthority
+		= Strat->WhoShouldHaveAuthority(*TestActors[ActorName]) == Strat->GetLocalVirtualWorkerId();
+	const bool DoWeActuallyHaveAuthority = Strat->ShouldHaveAuthority(*TestActors[ActorName]);
+
+	Test->TestEqual(
+		FString::Printf(TEXT("Expected: %b Actual: %b"), WeShouldHaveAuthority, DoWeActuallyHaveAuthority),
```

Comment:
Could have a string here before `Expected: x Actor: y` mentioning what the test is like you do in other places?

Reply:
added